storage/aptosdb/src/lib.rs

```diff
@@ -1380,10 +1381,7 @@ impl DbWriter for AptosDB {
             // Executing and committing from more than one threads not allowed -- consensus and
             // state sync must hand over to each other after all pending execution and committing
             // complete.
-            let _lock = self
-                .ledger_commit_lock
-                .try_lock()
-                .expect("Concurrent committing detected.");
+            let _lock = self.ledger_commit_lock.lock();
```

Comment:
Just double checked that `save_transactions_ext` is also used by state sync, and it appears so. So this looks good to me.